### PR TITLE
Change BlocklyToFeature to extend BlocklyToJava - Feature Mod element

### PR DESCRIPTION
**Pull request by Goldorion**
When I asked you [here ](https://github.com/MCreator/MCreator/pull/2528#discussion_r832614591) if it could use Blockly2Java instead of BlocklyToCode directly, you answered me you couldn't as you needed to execute something before the code execution.
I adapted a little bit (minor changes) `BlocklyToJava` and your `BlocklyToFeature`, so we can avoid repetitive code (the exact purpose of my recent changes with Blockly).

I did not add the `BlocklyEditorToolbar`, but it may be interesting to have it, so templates can be used in the future when complex features will be possible.

### CODE REVIEW

File: src/main/java/net/mcreator/ui/blockly/BlocklyEditorType.java [hunk 1]
```diff
@@ -22,7 +22,8 @@
 public enum BlocklyEditorType {
 
 	PROCEDURE("procedures", "ptpl", "event_trigger"),
-	AI_TASK("ai_setup", "aitpl", "aitasks_container");
+	AI_TASK("ai_setup", "aitpl", "aitasks_container"),
+	FEATURE("features", "ftpl", "feature_container");
```
Comment (AlsoSomeoneElse) [inline src/main/java/net/mcreator/ui/blockly/BlocklyEditorType.java]: While I do plan to add feature templates, I'd prefer to keep the first PR as small as possible. I'd revert these changes and just use "feature_container" for now

File: src/main/java/net/mcreator/blockly/feature/BlocklyToFeature.java [hunk 2]
```diff
@@ -19,86 +19,63 @@
...
-				List<Element> base_blocks = BlocklyBlockUtil.getBlockProcedureStartingWithNext(start_block);
-				if (base_blocks.isEmpty())
-					addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
-							L10N.t("blockly.errors.features.missing_placement")));
-				processBlockProcedure(base_blocks);
+				featureConfigurationCode.append(directProcessOutputBlock(this, feature));
+				Element featureBlock = XMLUtil.getFirstChildrenWithName(feature, "block");
+				if (featureBlock != null)
+					this.featureType = featureBlock.getAttribute("type");
 			} catch (TemplateGeneratorException e) {
```
Comment (AlsoSomeoneElse) [inline src/main/java/net/mcreator/blockly/feature/BlocklyToFeature.java]: I think it makes more sense to rethrow the exception, similar to what the `BlocklyToJava` constructor does

**Review by AlsoSomeoneElse — CHANGES_REQUESTED** [review]

File: src/main/java/net/mcreator/ui/blockly/BlocklyEditorType.java (hunk 1 above)
Comment (Goldorion) [reply src/main/java/net/mcreator/ui/blockly/BlocklyEditorType.java]: It was only to avoid more changes in the future and make the code already similar to everything else, so it has some consistency.

File: src/main/java/net/mcreator/blockly/feature/BlocklyToFeature.java (hunk 2 above)
Comment (Goldorion) [reply src/main/java/net/mcreator/blockly/feature/BlocklyToFeature.java]: I wasn't sure what to do for this part because to rethrow it, the general method has to throw the exception.

Comment (AlsoSomeoneElse) [reply src/main/java/net/mcreator/blockly/feature/BlocklyToFeature.java]: You could add `throws TemplateGeneratorException` to the method in `BlocklyToJava`. The override in `BlocklyToProcedure` doesn't need to be changed, since it doesn't throw anything

**Comment by AlsoSomeoneElse** [comment]
Looks good, thank you!
